src/main/kotlin/org/rust/ide/refactoring/inlineFunction/RsInlineFunctionProcessor.kt

```diff
@@ -158,296 +225,504 @@ class RsInlineFunctionProcessor(
+        val block = function.block ?: return
+        val declarationsByName = block.topLevelDeclarations
+        for ((name, declarations) in declarationsByName) {
+            val freshName = getFreshName(name) ?: continue
+            declarations.renameWithUsagesTo(freshName, function)
+        }
     }
 
-    private fun replaceParameterUsages(functionDup: RsFunction, caller: PsiElement, funcScope: LocalSearchScope) {
-        val parameterBindingsList = functionDup.valueParameters
-            .filter { it !is RsSelfParameter }
-            .map { it.pat }
-            .filterIsInstance(RsPat::class.java)
+    private fun RsElement.getAllVisibleBindingsIncludingContainingLetDeclaration(): Set<String> {
+        val declarationNames = ancestorStrict<RsLetDecl>()
+            ?.pat
+            ?.descendantsOfType<RsPatBinding>()
+            ?.map { it.referenceName }
+            .orEmpty()
+        return getAllVisibleBindings() + declarationNames
+    }
 
-        val valueArgumentList = when (caller) {
-            is RsCallExpr -> caller.valueArgumentList
-            is RsDotExpr -> caller.methodCall?.valueArgumentList ?: return
-            else -> return
+    // `Foo { field }` to `Foo { field: field }`
+    private fun expandStructLiteralFieldsShorthand(function: RsFunction) {
+        for (field in function.descendantsOfType<RsStructLiteralField>()) {
+            if (field.colon != null) continue
+            val name = field.identifier?.text ?: continue
+            field.add(factory.createColon())
+            field.add(factory.createExpression(name))
         }
+    }
 
-        val arguments = if (isDoubleSemicolonMethodCall(functionDup, caller)) {
-            valueArgumentList.exprList.drop(1)
-        } else {
-            valueArgumentList.exprList
+    // `Foo { field: field }` to `Foo { field }`
+    private fun collapseStructLiteralFieldsShorthand() {
+        for (field in function.descendantsOfType<RsStructLiteralField>()) {
+            val name = field.identifier?.text ?: continue
+            if (field.colon != null && name == field.expr?.text) {
+                RsFieldInitShorthandInspection.applyShorthandInit(field)
+            }
         }
+    }
+}
+
+private class InsertFunctionBody(
+    private val functionCall: RsCallExpr,
+    private val function: RsFunction,
+    private val letDeclarations: List<RsLetDecl>,
+) {
+
+    private val factory: RsPsiFactory = RsPsiFactory(function.project)
+    private lateinit var bodyStatements: List<RsElement>
+
+    fun invoke() {
+        val functionBody = function.block ?: return
+        if (functionBody.lbrace.getNextNonCommentSibling() == functionBody.rbrace) {
+            (functionCall.parent as? RsExprStmt)?.delete()
+            return
+        }
+
+        addLetDeclarationsToFunctionBody()
+        bodyStatements = functionBody.stmtsAndMacros.toList()
```

Comment:
We can't use `expanded...` here since `bodyStatements` are used to replace function call, so if function body has macros, we need to use it unexpanded. I don't look at `checkIfLoopCondition` in depth, but looks like there elements are used for some check and not for the insertion, so `expanded...` is ok there